### PR TITLE
Redesign Duration display for Dutch auctions with live-updating price visualization.

### LINKED ISSUE #1088: Redesign "duration" field in auction view to show a live-updating slider
We know an auction's start/end heights and step count from `DutchAuctionDescription`. We know the current block height from the `status` endpoint. From these pieces of data, we can show a live-updating slider on the view of an auction to depict how far along the auction is toward completion:

$`(currentHeight - startHeight) / (endHeight - startHeight)`$

We can also deduce how many blocks are between each step, in case we want to e.g. show a dot on the graph for each step:

$`(endHeight - startHeight) / stepCount`$

This ticket is to render a UI using this data.

Algorithm for price at each step:

https://github.com/penumbra-zone/penumbra/blob/e0291fb/crates/core/component/auction/src/component/dutch_auction.rs#L548-L572

### PR DESCRIPTION
**Pull request by jessepinho**
Scrub through this video to get a timelapse of part of a 10-minute auction.

https://github.com/penumbra-zone/web/assets/1121544/ad04a0cf-3020-4560-939d-0bd69a15020f

This is a bit of a frilly PR that I prob shouldn't have spent as much time on as I did... Basically it visualizes to the user what the current price is of the input asset in terms of the output asset, as well as how far along we are in the auction.

Again, this is by no means a finalized design, but it's at least an improvement on what we had before.

## In this PR
- Design a live-updating display for Dutch auctions.
- Create a new Zustand `status` slice for storing the sync status (`fullSyncHeight` and `latestKnownBlockHeight`) so it can be used both in the header and in places like Dutch auctions.
- Simplify `<Progress />` to only accept a few specific props, rather than the full default set (which we neither need nor want).
- Add size variants to both `<AssetIcon />` and `<ValueViewComponent />`.

Closes #1088

### CODE REVIEW

File: packages/ui/lib/toast/transaction-toast.test.tsx [hunk 1]
```diff
@@ -67,13 +67,9 @@ describe('TransactionToast', () => {
...
         expect.objectContaining({
           description: (
-            <Progress
-              status='in-progress'
-              background='stone'
-              value={50}
-              size='sm'
-              className='mt-2'
-            />
+            <div className='mt-2'>
```
Comment (jessepinho) [inline packages/ui/lib/toast/transaction-toast.test.tsx:70]: I removed `className` from `<Progress />`'s props, so we need to wrap it in a div with a margin-top.

File: apps/minifront/src/components/Status.tsx
Comment (grod220) [inline apps/minifront/src/components/Status.tsx:1]: Ah! Another case for https://github.com/penumbra-zone/web/issues/965.

File: packages/ui/components/ui/dutch-auction-component/get-price.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  // The target output, scaled up by `step_count` to avoid divisions.
+  // Linearly interpolate between `max_output` at `step_index = 0`
+  //                          and `min_output` at `step_index = step_count - 1`.
+  const targetOutputScaled = (stepCount - stepIndex - 1n) * maxOutput + stepIndex * minOutput;
+  // The input, scaled up by `step_count` to match.
+  const inputScaled = (stepCount - 1n) * input;
+
+  // The trading function interpolates between (input, 0) and (0, target_output)
+  const p = targetOutputScaled;
+  const q = inputScaled;
```
Comment (grod220) [inline packages/ui/components/ui/dutch-auction-component/get-price.ts:50]: Do we need these reassigned? Can we just use `targetOutputScaled` and `inputScaled` inline below?

File: packages/ui/components/ui/dutch-auction-component/get-progress.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const clampToDecimal = (value: number) => Math.min(Math.max(value, 0), 1);
```
Comment (grod220) [inline packages/ui/components/ui/dutch-auction-component/get-progress.ts:1]: What does clamp mean here? Maybe a code comment here? (same in other usage too)

**Review by grod220 — APPROVED** [review]
Nice! Looks good.

File: packages/ui/components/ui/dutch-auction-component/get-price.ts (hunk 2 above)
Comment (jessepinho) [reply packages/ui/components/ui/dutch-auction-component/get-price.ts]: Ah, good catch — this was a leftover from copying core's implementation.

File: packages/ui/components/ui/dutch-auction-component/get-progress.ts (hunk 3 above)
Comment (jessepinho) [reply packages/ui/components/ui/dutch-auction-component/get-progress.ts:1]: [Clamping](https://en.wikipedia.org/wiki/Clamping_(graphics)) refers to limiting a value between a given min and max. Will add a comment.

**Pull request merged**